### PR TITLE
add pkg scheme to npm generic file PURLs

### PR DESCRIPTION
**Pull request by taylormadore**
The pkg scheme is required for each component according to the PURL specification.

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [n\a] New code has type annotations
- [n\a] OpenAPI schema is updated (if applicable)
- [n\a] DB schema change has corresponding DB migration (if applicable)
- [n\a] README updated (if worker configuration changed, or if applicable)
- [ ] Draft release notes are updated before merging

### CODE REVIEW

**Comment by eskultety** [comment]
LGTM

File: tests/integration/test_data/npm_packages.yaml [hunk 1]
```diff
@@ -782,22 +782,22 @@ workspaces:
...
-      - "generic/bar?file%3Abar"
-      - "generic/eggs?file%3Aeggs-packages%2Feggs"
-      - "generic/foo?file%3Afoo"
-      - "generic/not-baz?file%3Abaz"
-      - "generic/spam?file%3Aspam-packages%2Fspam"
+      - "pkg:generic/bar?file%3Abar"
+      - "pkg:generic/eggs?file%3Aeggs-packages%2Feggs"
+      - "pkg:generic/foo?file%3Afoo"
+      - "pkg:generic/not-baz?file%3Abaz"
+      - "pkg:generic/spam?file%3Aspam-packages%2Fspam"
```
Comment (chmeliik) [inline tests/integration/test_data/npm_packages.yaml:789]: I'm fairly certain these are still invalid purls (even syntactically) :disappointed: 

See qualifiers in https://github.com/package-url/purl-spec/blob/master/PURL-SPECIFICATION.rst#rules-for-each-purl-component. TLDR they must be key=value pairs

Comment (brunoapimentel) [reply tests/integration/test_data/npm_packages.yaml:789]: Just my 2¢. If it's easy to fix the qualifiers to be `key=value`, let's do it in this PR, but let's not try to make the PURLs perfect, as they're already broken now, and we're not making them worse.

We can file a follow up story for any other fixes that are needed, and we can work on it when we have time :man_shrugging:

Comment (taylormadore) [reply tests/integration/test_data/npm_packages.yaml:789]: I'll revisit this in the morning :+1: 
Just to confirm, we are OK with a PURL like this: `pkg:generic/%40jaegertracing/plexus?file=packages%2Fplexus`

**Review by brunoapimentel — APPROVED** [review]

**Pull request merged**